pkg/adaptor/proxy/proxy.go

```diff
@@ -78,8 +103,13 @@ func (p *agentProxy) dial(ctx context.Context, address string) (net.Conn, error)
 			ctx, cancel := context.WithTimeout(ctx, p.retryInterval)
 			defer cancel()
 
-			// TODO: Support TLS
-			conn, err = (&net.Dialer{}).DialContext(ctx, "tcp", address)
+			if dialer.Config == nil {
+				// Use generic tcp dialer
+				conn, err = dialer.NetDialer.DialContext(ctx, "tcp", address)
+			} else {
+				// Use tls dialer
+				conn, err = dialer.DialContext(ctx, "tcp", address)
+			}
```

Comment:
Then, these line will be like this.
```
conn, err = dialer.DialContext(ctx, "tcp", address)
```

Reply:
Thanks for the suggestion. I'll try this and let you know.

Reply:
I tested the changes on libvirt using both tls and plain connection